Bad dunder name is flagging __index__

### Bug description

consider 
```
class X:
    def __index__(self):
        return 1
```

this will flag [__index__](https://docs.python.org/3/library/operator.html#operator.__index__) as a bad dunder, even though it is valid to do.

### Configuration

_No response_

### Command used

```shell
pylint x.py
```


### Pylint output

```shell
************* Module x
x.py:2:4: W3201: Bad or misspelled dunder method name __index__. (bad-dunder-name)
```


### Expected behavior

no lint

### Pylint version

```shell
% pylint --version
pylint 2.17.2
astroid 2.15.4
Python 3.10.11 (main, Apr 10 2023, 08:08:57) [GCC 8.3.0]
```


### OS / Environment

_No response_

### Additional dependencies

_No response_

--- a/doc/whatsnew/fragments/8613.false_positive
+++ b/doc/whatsnew/fragments/8613.false_positive
@@ -1,0 +1,3 @@
+Fix a false positive for ``bad-dunder-name`` when there is a user-defined ``__index__`` method.
+
+Closes #8613

--- a/pylint/constants.py
+++ b/pylint/constants.py
@@ -231,6 +231,7 @@ EXTRA_DUNDER_METHODS = [
     "__getnewargs_ex__",
     "__getnewargs__",
     "__getstate__",
+    "__index__",
     "__setstate__",
     "__reduce__",
     "__reduce_ex__",

--- a/tests/functional/ext/bad_dunder/bad_dunder_name.py
+++ b/tests/functional/ext/bad_dunder/bad_dunder_name.py
@@ -49,6 +49,9 @@ class Apples:
     def __doc__(self):
         return "Docstring"
 
+    def __index__(self):
+        return 1
+
 
 def __increase_me__(val):
     return val + 1

